### ISSUE TITLE
New summary and description

### ISSUE THREAD

**Issue by meh** [meh/smart-referer#70]
All the 0.2 versions have been disabled by an AMO reviewer who didn't spend more than 5 seconds looking at the summary, description and the addon (he even requested screenshots :rage4:).

So we need a "better" one, I'm too pissed to do it right now, if anyone wants to, go ahead.

**Comment by antstuff** [comment]
Arghhh...

I just installed Firefox 57 fresh, after upgrading 56 and seeing the smart-referer worked flawlessly, and now I can't install smart-referer on Firefox 57, mozilla only offers version 0.0.11x... I am so pissed.

Is there another place where I can download the 0.2x version xpi ?

I left a message on mozilla's firefox add-on forums asking to reactivate smart-referer 0.2 version... hope they will do something.

What do they want exactly ? A screenshot ?

**Comment by meh** [comment]
>1) Please make your add-on listing more elaborate. Your summary should briefly explain what the add-on does, so that even people without knowledge about your add-on and its services can understand. The description should give details on the add-on features, how to configure and if not apparent then how to use it. The listing should also contain one or more screenshots.

I wouldn't be as pissed if he didn't disable all the web extension versions right before the release of 57, and with a request that clearly shows he didn't even look at the addon.

**Comment by antstuff** [comment]
Yes... Grrr... 

Now I see the downsides to having extensions in a store-"like" location like that.... it seems weird that an add-on can be removed so easily on a whim...

**Comment by scribblemaniac** [comment]
While it's stupid that they should disable the web extension over such a detail, they do have a point. The description is only a single sentence, and if someone does not know about the Referer header and what it does, then they would have absolutely no idea from your description what the purpose of this addon is. Bear in mind the vast majority of Internet users don't know anything about variables, headers, HTTP, JavaScript, etc. but they could still benefit from this extension.

**Comment by meh** [comment]
@scribblemaniac I entirely agree, I'm just extremely pissed at how it was handled and I'm already fucking mad at them for dropping XUL support since it's the only reason I even use the browser instead of writing my own.

On Firefox ESR with GTK2 right now, already feeling grumpy, that just made me fucking mad.

**Comment by antstuff** [comment]
Yes... it's a fair point... but it still is a form of regulation which I
don't think I support... If I don't know what a plug-in does, or I am not
conformable with it's description, I just don't install it, it's up to me
to make that choice, not up to mozilla to make that choice for me... I
don't like for these kind of choices being made for me..

So if you put the same description as the 0.11 version they still allowed,
just adding "Webextension version comptaible with Firefox >57" it should
bee fine shouldn't it ?

2017-11-14 23:12 GMT+01:00 scribblemaniac <notifications@github.com>:

**Comment by antstuff** [comment]
I'm mad that it was removed, and I'm just a user... so I can perfectly
image how mad you must be as the dev.. lol

Thank you for making it available for us as webextension anyway, I really
appreciate your work, thank you

2017-11-14 23:20 GMT+01:00 ant feeds <antfeeds@gmail.com>:

**Comment by meh** [comment]
@antstuff the person to really thank is @alexander255, I actually just reviewed the web extension code.

@alexander255 if you want to take the whole thing I'll just add you to AMO if you can tell me how, I don't think I'm going to keep using Firefox for long anyway.

**Comment by antstuff** [comment]
Thank you both then :)

**Comment by kah0922** [comment]
You could try just copying the Github README file and use that as the description.

**Comment by alexander255** [comment]
@meh: Could you maybe post their exact message they've sent you? (Or forward me the message via e-mail.)
Also if you add me on AMO I'll add the description and screenshots myself (similar to the description of [User-Agent Switcher](https://addons.mozilla.org/de/firefox/addon/uaswitcher/)).

**Comment by alexander255** [comment]
(Updated comment to reflect changes in this thread since between the time I read it and posted my inital response.)

**Comment by meh** [comment]
@alexander255 done.

**Comment by alexander255** [comment]
@meh Thanks I'll look into appleasing Mozilla in the next day or two.

@antstuff: For now I published a signed release here: https://github.com/meh/smart-referer/releases/tag/v0.2.4

**Comment by antstuff** [comment]
@alexander255 Sweet :) Works like a charm, thank you

**Comment by kah0922** [comment]
[Looks like it's back on AMO](https://addons.mozilla.org/en-US/firefox/addon/smart-referer/)

**Comment by antstuff** [comment]
Yes, i'm happy as a puppy, see : https://discourse.mozilla.org/t/solved-smart-referer-0-2-by-meh-removed-please-put-it-back/21610

I guess it will be updated to version v0.2.4 soon, might be a good idea to update the description with the github readme text, which is very clear, anyway...

**Comment by alexander255** [comment]
@antstuff: Thanks for getting this reported at Mozilla's, looking at the history of message exchanges between meh and the Mozilla reviewer, I kinda understand why meh may have pissed him off (althrough his reaction was still inappropriate) 😁 
Version 0.2.4 was a private release (with no changes compared to 0.2.3) so that we could continue distribution of the extension until we figure out the situation with Mozilla on this. It will not be publically listed on AMO.

**Comment by antstuff** [comment]
Ok, cool

2017-11-16 12:24 GMT+01:00 Alexander Schlarb <notifications@github.com>:

**Comment by aschrijver** [comment]
Just like @scribblemaniac [wrote](https://github.com/meh/smart-referer/issues/70#issuecomment-344416487) I fully agree the summary and description, both in the extension page _and_ in the README should contain some introductory text and explanation in layman's terms on why you would even consider fiddling with your referer.

This is a great addon, but currently only interesting to people 'in the know', technical people. With growing awareness of privacy issues, it is important to get the broader public on-board, including tech noobs.

**Comment by alexander255** [comment]
@aschrijver: True! Mind writing something up and pasting it here for “peer review”. This has been on my todo list for a long time and it could really use that boost. :slightly_smiling_face:

**Comment by ntninja** [comment]
This issue has been moved to GitLab: https://gitlab.com/smart-referer/smart-referer/issues/70
